lib/matchers/email.js

```diff
@@ -0,0 +1,64 @@
+var escapeStringRegexp  = require('escape-string-regexp');
+var utils               = require('../utils');
+var factory             = require('../factory');
+
+var TYPE_ERROR = 'should be a valid email address';
+
+var DOMAIN_REGEX_SOURCE = /[a-zA-Z0-9](-?[a-zA-Z0-9])*(\.[a-zA-Z](-?[a-zA-Z0-9])*)+/.source;
+var DOMAIN_REGEX = new RegExp('^' + DOMAIN_REGEX_SOURCE + '$');
+var EMAIL_REGEX_SOURCE = /[-!#$%&'*+\/0-9=?A-Z^_a-z{|}~](\.?[-!#$%&'*+/0-9=?A-Z^_a-z`{|}~])*@/.source;
+var EMAIL_REGEX = new RegExp('^' + EMAIL_REGEX_SOURCE + DOMAIN_REGEX_SOURCE + '$');
+
+// Stolen from http://thedailywtf.com/articles/Validating_Email_Addresses
+
+module.exports = factory({
+  initialize: function(opts) {
+    this.opts = opts || {};
+
+    this.hasDomain = utils.hasValue(this.opts.domain);
+
+    if (this.hasDomain && !DOMAIN_REGEX.test(opts.domain)) {
+      throw new Error('Invalid domain value: ' + opts.domain);
+    }
+
+    if (this.hasDomain) {
+      this.domainRegex = this.regex = new RegExp('^' + EMAIL_REGEX_SOURCE + escapeStringRegexp(opts.domain) + '$');
+    }
+    else
+      this.regex = EMAIL_REGEX;
+  },
+
+  match: function(path, value) {
+    if (typeof value !== 'string') return TYPE_ERROR;
+
+    if (value.length > 254)
+      return TYPE_ERROR;
+
+    if (!EMAIL_REGEX.test(value))
+      return TYPE_ERROR;
```

Comment:
What I don't understand is consistency! why we should have three different types of if return syntax in a file, two of them in same function!

You have
```javascript
if (typeof value !== 'string') return TYPE_ERROR;
```

and you have
```javascript
if (value.length > 254)
      return TYPE_ERROR;
```

Also generally speaking this syntax is not accepted in javascript
```javascript
if (value.length > 254)
      return TYPE_ERROR;
```

Take a look at other area of strummer codebase, you will see how if return is used, consistency matters.